cpp/src/io/parquet/page_enc.cu

```diff
@@ -206,7 +206,8 @@ void __device__ calculate_frag_size(frag_init_state_s* const s, int t)
     }
   }
   __syncthreads();
-  auto const total_len   = block_reduce(reduce_storage).Sum(len);
+  auto const total_len = block_reduce(reduce_storage).Sum(len);
+  __syncthreads();
```

Comment:
So we really need two sync like this?

Reply:
hm, maybe we should remove it if racecheck does not complain

Reply:
I'll try removing it and see

Reply:
> I'll try removing it and see
trying this on my end as well

> What is racecheck?
racecheck tool in compute-sanitizer

Reply:
remove?

Reply:
no racecheck errors, and, looking at the code, no shared data is modified before the reductions. IMHO we should be good to remove the first sync.